src/java.base/share/classes/sun/security/action/GetPropertyAction.java

```diff
@@ -224,4 +224,37 @@ public static int privilegedGetTimeoutProp(String prop, int def, Debug dbg) {
             return def;
         }
     }
+
+    /**
+     * Convenience method for fetching System property values that are booleans.
+     *
+     * @param prop the name of the System property
+     * @param def a default value
+     * @param dbg a Debug object, if null no debug messages will be sent
+     *
+     * @return an boolean value corresponding to the value in the System property.
+     *      If the property value is neither "true" or "false", the default value
+     *      will be returned.
+     */
+    public static boolean privilegedGetBooleanProp(String prop, boolean def, Debug dbg) {
```

Comment:
It probably makes more sense to put this method in the `GetBooleanAction` class.

Reply:
I thought so first too, but decided that `GetPropertyAction` is a better place for it, because it needs the generic `privilegedGetProperty`. It is also adjacent to `privilegedGetTimeoutProp` that returns `int`. Makes sense?

Reply:
Right, that's a good point because you need to get the string value of the property and not a boolean directly. Ok then.

For a test, what do you think about adding another @run line to `GetAndPostTest` like:

`* @run main/othervm -Dcom.sun.security.ocsp.useget=treu GetAndPostTests`

and also modifying the POST code of test/jdk/java/security/testlibrary/SimpleOCSPServer.java to check that when the request is less than 255 bytes, the system property is set to "false" and nothing else? The server runs in the same process as the test, so I think that should work.

This would help eliminate the possibility of the property accidentally not working in the future and reverting to GET instead of POST.

Reply:
This looks easy/safe enough to do. See new commits. New version passes the `GetAndPostTests` and the whole `jdk_security`. For sensitivity test, I broke the option parsing and observed that `GetAndPostTests` fails as expected.